source/backend/api/Services/PropertyService.cs

```diff
@@ -209,9 +209,70 @@ public PropertyManagementModel GetPropertyManagement(long propertyId)
             propertyManagement.RelatedLeases = leaseCount;
             propertyManagement.LeaseExpiryDate = leaseExpiryDate.HasValue ? DateOnly.FromDateTime(leaseExpiryDate.Value) : null;
 
+            PropertyHasActiveLease(propertyLeases, out bool hasActiveLease, out bool hasActiveExpiryDate);
+            propertyManagement.HasActiveLease = hasActiveLease;
+            propertyManagement.ActiveLeaseHasExpiryDate = hasActiveExpiryDate;
+
             return propertyManagement;
         }
 
+        private static void PropertyHasActiveLease(IEnumerable<PimsPropertyLease> propertyLeases, out bool hasActiveLease, out bool hasActiveExpiryDate)
+        {
+            hasActiveLease = false;
+            hasActiveExpiryDate = false;
+
+            List<PimsLease> activeAgreementsList = propertyLeases.Select(x => x.Lease).ToList();
+            var activeLease = activeAgreementsList.FirstOrDefault(x => x.LeaseStatusTypeCode == LeaseStatusTypes.ACTIVE.ToString());
+
+            // No Active Lease
+            if (activeLease is not null)
+            {
+                // Lease is Active but has termination Date.
+                if (activeLease.TerminationDate.HasValue)
+                {
+                    return;
+                }
+                else
+                {
+                    var latestRemewal = activeLease.PimsLeaseRenewals.Where(x => x.IsExercised.HasValue && x.IsExercised.Value).OrderByDescending(x => x.CommencementDt).FirstOrDefault();
+                    if (latestRemewal is null) // No Renewal - Check only Lease dates.
+                    {
+                        if(activeLease.OrigExpiryDate.HasValue && activeLease.OrigExpiryDate.Value.Date >= DateTime.UtcNow.Date)
```

Comment:
I think this is an example of when you don't want to use the UTC date. If you did, and a user used this logic before 7am, the utc date would not match the pst date.

Reply:
Another way to think of this - when we store dates with times, we store them in UTC. When we store dates only, we store them as-is.

Reply:
Please note that this is backend (C#) code so this runs in the timezone of the .NET container - which by default is UTC in docker/openshift. I'm not aware that we set the timezone to PST on our live env containers.
If we want to do this "Date math" in PST we need to do that manually (either subtract 7hrs from UTC or convert the exp date to UTC)

Reply:
Updated